enhancements/upgrade-bundle.md

```diff
@@ -0,0 +1,260 @@
+2. Should we explicitly pass new version number as a flag or can we get it from bundle image?
+    * Based on discussions, can extract the version number from bundle image. 
+
+## Summary
+
+Enhance the Operator SDK to add a new subcommand `run bundle-upgrade` that performs an operator upgrade using OLM.
+
+## Motivation
+
+Testing operator upgrades remains difficult today due to the number of steps to prepare for an upgrade and the lack of tooling that automates those steps. Having high quality Operators is crucial to the success of the Operator Framework and thereby operator developers should have a tool to use to perform an operator upgrade.
+
+### Goals
+
+Provide a command line that upgrades an operator version using OLM.
+
+### Non-Goals
+
+This enhancement doesn't include the capability for SDK to use labels to detect the DB path. For the sake of simplicity, we will assume that the existing registry DB is located at `/database/index.db`.
+
+Another non-goal is that the enhancement is intended for operators that don't exist in the cluster. The scenario in which existing operators exist in the cluster is out of scope for this proposal.
+
+## Proposal
+
+### Implementation Details/Notes/Constraints [optional]
+
+The Operator SDK currently supports running an operator with OLM using the
+`run packagemanifests` subcommand. 
+This enhancement will add a new subcommand, `bundle-upgrade`,
+to the `run` command. This new command, `run bundle-upgrade`, will perform the upgrade of an operator from a previous version to a new specified version.
+
+Currently, the run command has different subcommands: 
+
+ * `run local`
+ * `run packagemanifests`
+ * `run bundle` - currently being implemented, design document is covered by [run bundle enhancement](https://github.com/operator-framework/enhancements/blob/master/enhancements/run-bundle.md)
+
+For doing an `bundle-upgrade`, there could be two scenarios: 
+
+* Case 1 - A previous version of the operator is already running on the cluster, and the index image contains this operator version. Now, we we want to upgrade the in cluster running operator to a new version that we specify. 
+
+* Case 2 - No previous version of an operator is running on the cluster, so we provide the new bundle image which is accessible and pullable within the cluster, which contains our desired version of the operator for the upgrade. 
+
+In case 1, we will have to figure out which catalog source the existing operator was installed in and somehow dynamically inject the new bundle image into the index image for OLM to detect this change and upgrade to the new version we provided. 
```

Comment:
Actually, OLM is capable of resolving over multiple catalogs. IIRC, you can just create a new index that contains a replacement bundle in the same package/channel as the current `Subscription` and OLM will perform the upgrade for you (should the `Subscription` be on "automatic approval").

Reply:
This seems totally fine. But in production, can we create a new index while the previous version of the operator (with the old index) is already running in the cluster and dynamically modify the image reference to point to this new index image we just built? Is that possible?

Reply:
If we set the approval to `automatic`, OLM always updates to the latest version available in the channel we specify in this command, in which case testing the operator version we wanted to upgrade to, might be difficult. Thoughts?